### ISSUE TITLE
pureconfig-magnolia: support reading/writing case objects

### ISSUE THREAD

**Issue by neko-kai** [pureconfig/pureconfig#796]
Currently `pureconfig-magnolia` reads/writes case objects only as empty objects instead of strings, e.g

```scala
sealed trait SealedTrait
object SealedTrait {
  case object Yes extends SealedTrait
  case object No extends SealedTrait
}
```

Becomes

```
{ No {} }
```

in config instead of

```
"No"
```

**Comment by ruippeixotog** [comment]
The expected default behavior of sealed traits is to disambiguate using a `type` field, so `No` would result in `{ type: "no" }`. Can you share some minimal example in which `{ No {} }` is written or read by PureConfig?

To achieve the behavior you want you need to use `deriveEnumerationReader`. Please take a look at https://pureconfig.github.io/docs/overriding-behavior-for-sealed-families.html.

**Comment by neko-kai** [comment]
> Can you share some minimal example in which { No {} } is written or read by PureConfig?

Ah, I've added a custom coproduct hint to read sealed traits in the style of `circe`, and expose auto-derivations using it in `distage-config` library:
https://github.com/7mind/izumi/blob/develop/distage/distage-extension-config/src/main/scala/izumi/distage/config/codec/PureconfigInstances.scala
https://github.com/7mind/izumi/blob/develop/distage/distage-extension-config/src/main/scala/izumi/distage/config/codec/PureconfigAutoDerive.scala

I've forgot this isn't pureconfig's default behavior.

> To achieve the behavior you want you need to use deriveEnumerationReader. Please take a look at https://pureconfig.github.io/docs/overriding-behavior-for-sealed-families.html.

That's correct, but this instance is part of `pureconfig-generic`, there isn't a way to derive an enumeration reader using only the magnolia module yet.

**Comment by ruippeixotog** [comment]
Right, I forgot that we didn't have enumeration readers and writers for Magnolia. Sounds like a good feature to add!

**Comment by jcazevedo** [comment]
Closed by #798.
